server/src/completion/YmlCompletionItemsProvider.ts

```diff
@@ -8,11 +8,28 @@ export class YmlCompletionItemsProvider {
     }
 
     /**
-     * Add a completion to this provider.
+     * Add a completion to this provider. Also keep track of each of its label subparts (separated with `:`).
+     *
+     * For example, adding a function named `Namespace:Type::FunctionName`
+     * will add to the suggestions list the function as well as the tokens `Namespace` `Type` and `FunctionName`.
+     *
      * @param complLoc The new completion to add.
      */
     public addCompletionItem(complLoc: AbstractYmlObject): void {
         this.completions.push(complLoc);
+        const subElements = complLoc.label.split(/::?/);
+        /* Single element. This has already been added. */
+        if (subElements.length === 1) {
+            return;
+        }
+        /* Keep all subparts of the object's label, even its short name. */
+        for (const word of subElements) {
```

Comment:
I'd check for non-blankness, since a typo could lead to a `:::::::`, which would make `split` yield empty elements.
That, or turn the regex into `:+`.

Reply:
Done.

Reply:
I realized a bit late that the `:+` technique can still produce empty elements if the colons stand at the end of the string (i.e., when we haven't typed what follows yet). However:

* maybe you only call this method with things that match a pattern avoiding this?
* I'm not even sure it's _that_ awful to add empty elements.

Still, to avoid unnecessary questions and make things look more… natural, I'd probably use the `:+` _and_ a `if not blank` together, if I were you.

Reply:
- Yes, indeed, I'm calling this method only when we encounter an YML object that has a correct identifier.
Identifiers cannot finish with a colon, it should be OK.
- I think this is awful :smile: 

I will add a `if not blank` condition on the `word` variable.